spartansw/argo-drupal-module
# [Argo] Add support for config/locale translations

## Release Notes
- New API endpoint to export config strings for translations (i.e. `/argo/config/{langcode}/export`). Additional option to include/exclude config values with existing translations can be controller via query parameter `include-translations=0/1`.
- New API endpoint to import config translations (i.e. `/argo/config/{langcode}/translate`).
- New API endpoint to export UI/locale strings for translations (i.e. `/argo/locale/{langcode}/export`). Additional option to include/exclude UI strings with existing translations can be controller via query parameter `include-translations=0/1`.
- New API endpoint to import UI/locale string translations (i.e. `/argo/locale/{langcode}/translate`).

## QA
1. Export config strings
http://argo-tableau-www.pantheonsite.io/argo/config/fr-FR/export
- [ ] Assert already translated config strings are excluded
http://argo-tableau-www.pantheonsite.io/argo/config/fr-FR/export?include-translations=1
- [ ] Assert the number is larger and all config strings - irregardless of their translations status - are included.

2. Import config translations
_Note: Use Postman for this_

Request: `POST`
Url: `http://argo-tableau-www.pantheonsite.io/argo/config/fr-FR/translation`
Body (JSON):
```
[
    {
        "key": "elements.phone.#title",
        "string": "Phone",
        "translation": "Phone FR",
        "context": "@todo",
        "url": "@todo",
        "config_id": "webform.webform.online_trial"
    },
    {
        "key": "handlers.tableau_online_trial.label",
        "string": "Tableau Online Trial",
        "translation": "Tableau Online Trial Fr-FR",
        "context": "@todo",
        "url": "@todo",
        "config_id": "webform.webform.online_trial"
    }
]
```
- [ ] Check the translations via http://argo-tableau-www.pantheonsite.io/admin/structure/webform/manage/online_trial/translate/fr-FR/edit

3. Export UI strings
 http://argo-tableau-www.pantheonsite.io/argo/locale/fr-FR/export
- [ ] Assert already translated UI strings are excluded
http://argo-tableau-www.pantheonsite.io/argo/locale/fr-FR/export?include-translations=1
- [ ] Assert the number is larger and all UI strings - irregardless of their translations status - are included.

4. Import UI string translations
Request: `POST`
Url: `http://argo-tableau-www.pantheonsite.io/argo/locale/fr-FR/translation`
Body (JSON):
```
[
    {
        "string": "additional",
        "translation": "additional FR",
        "context": "Aria Label",
        "url": "/es-es/learn/webinars/prep-conductor-making-data-prep-work-scale-your-organization"
    }
]
```
- [ ] Check the translation via http://argo-tableau-www.pantheonsite.io/admin/config/regional/translate

![Screen Shot 2021-04-02 at 11 26 22 AM](https://user-images.githubusercontent.com/8611594/113443169-47991100-93a6-11eb-86bc-7cf04412db22.png)
